src/library/dtls_test.cpp

```diff
@@ -42,87 +42,114 @@ namespace ot {
-                                       "HQ8EBAMCBeAwbAYDVR0RBGUwY6RhMF8xCzAJBgNVBAYTAlVTMRUwEwYDVQQKDAxU\r\n"
-                                       "aHJlYWQgR3JvdXAxFzAVBgNVBAMMDlRlc3QgUmVnaXN0cmFyMSAwHgYJKoZIhvcN\r\n"
-                                       "AQkBFhFtYXJ0aW5Ac3Rva29lLm5ldDBHBgNVHSMEQDA+gBSS6nZAQEqPq08nC/O8\r\n"
-                                       "N52GzXKA+KEcpBowGDEWMBQGA1UEAwwNVGhyZWFkR3JvdXBDQYIIc5C+m8ijatIw\r\n"
-                                       "CgYIKoZIzj0EAwIDSAAwRQIgbI7Vrg348jGCENRtT3GbV5FaEqeBaVTeHlkCA99z\r\n"
-                                       "RVACIQDGDdZSWXAR+AlfmrDecYnmp5Vgz8eTyjm9ZziIFXPUwA==\r\n"
+                                       "MIIBdjCCARygAwIBAgIUTsd8PPWTr5Dl8P1jj8V3tlmDGDowCgYIKoZIzj0EAwIw\r\n"
+                                       "GDEWMBQGA1UEAwwNVGhyZWFkR3JvdXBDQTAeFw0yNTA5MjQwNDUxMDlaFw0zNTA5\r\n"
+                                       "MjIwNDUxMDlaMBoxGDAWBgNVBAMMD1RocmVhZFJlZ2lzdHJhcjBZMBMGByqGSM49\r\n"
+                                       "AgEGCCqGSM49AwEHA0IABGr5hdFY+5eaF1vhw6wG+0Mybw0tauCxG04X7OqXv7/P\r\n"
+                                       "V7Y+teABvZkorhF2b332Z7Pqk/6k+wuCX1N5VAZJtyijQjBAMB0GA1UdDgQWBBQf\r\n"
+                                       "ASGIrYEzMd1F/eYF1IzmZ5M5bTAfBgNVHSMEGDAWgBTOYzG13fhmKHoLhpg11pkQ\r\n"
+                                       "KOBBXzAKBggqhkjOPQQDAgNIADBFAiEA4zluVAVVDfsCCuv4OSwx9o1P7w+QvmEC\r\n"
+                                       "xhJPt7eGQRYCIEgzvrcQ4VPinEe8t3CkIrrHe/zQrkHw9ZhQnLv509XW\r\n"
                                        "-----END CERTIFICATE-----\r\n";
 
-static const std::string kServerKey = "-----BEGIN PRIVATE KEY-----\r\n"
-                                      "MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQgYJ/MP0dWA9BkYd4W\r\n"
-                                      "s6oRY62hDddaEmrAVm5dtAXE/UGhRANCAAQgMIVb6EaRCz7LFcr4Vy0+tWW9xlSh\r\n"
-                                      "Xvr27euqi54WCMXJEMk6IIaPyFBNNw8bJvqXWfZ5g7t4hj7amsvqUST2\r\n"
-                                      "-----END PRIVATE KEY-----\r\n";
+static const std::string kServerKey = "-----BEGIN EC PARAMETERS-----\r\n"
+                                      "BggqhkjOPQMBBw==\r\n"
+                                      "-----END EC PARAMETERS-----\r\n"
+                                      "-----BEGIN EC PRIVATE KEY-----\r\n"
+                                      "MHcCAQEEIMdyKql6JZNPhCw+KSY/zbYKRor5qoebmt9kQQ73MzWcoAoGCCqGSM49\r\n"
+                                      "AwEHoUQDQgAEavmF0Vj7l5oXW+HDrAb7QzJvDS1q4LEbThfs6pe/v89Xtj614AG9\r\n"
+                                      "mSiuEXZvffZns+qT/qT7C4JfU3lUBkm3KA==\r\n"
+                                      "-----END EC PRIVATE KEY-----\r\n";
 
 static const char        *kServerAddr = "::";
 static constexpr uint16_t kServerPort = 5683;
 
-TEST(DtlsTest, MbedtlsClientServer)
+// Helper function to set up client configuration
+void SetUpClientConfig(DtlsConfig &config)
```

Comment:
nit: can the server hostname configuration being added to this SetUpClientConfig() method as an optional parameter? For example:

```
void SetUpClientConfig(DtlsConfig &config, const std::string& peerHostname="ThreadRegistrar")
```

Because it looks like all the DTLS tests which doesn't specifically test against the hostname verification need to manually set the hostname after calling SetUpClientConfig(). This is repeated and easy to be missed.